### PR TITLE
record restriction grace period: add tombstone page

### PR DESCRIPTION
**Pull request by anikachurilova**
* closes https://github.com/inveniosoftware/product-rdm/issues/178

Tombstone page for record restriction:
Screenshot 2024-05-15 at 09 37 20

### CODE REVIEW

File: invenio_app_rdm/theme/assets/semantic-ui/less/invenio_app_rdm/theme/elements/icon.overrides [hunk 1]
```diff
@@ -28,3 +28,7 @@
     z-index: 1;
   }
 }
+
+.right-floated {
+  float: right;
```
Comment (kpsherva) [inline invenio_app_rdm/theme/assets/semantic-ui/less/invenio_app_rdm/theme/elements/icon.overrides:33]: we already have this class defined in site.overrides, most likely it just misses `!important` clause if it didn't take effect on the icon

File: invenio_app_rdm/records_ui/views/records.py [hunk 2]
```diff
@@ -412,8 +412,24 @@ def record_tombstone_error(error):
 
 
 def record_permission_denied_error(error):
-    """Handle permission denier error on record views."""
+    """Handle permission denied error on record views."""
     if not current_user.is_authenticated:
         # trigger the flask-login unauthorized handler
         return current_app.login_manager.unauthorized()
+
+    record = getattr(error, "record", None)
```
Comment (kpsherva) [inline invenio_app_rdm/records_ui/views/records.py:420]: see my comments in the other PR

File: invenio_app_rdm/records_ui/views/records.py [hunk 3]
```diff
@@ -412,8 +412,24 @@ def record_tombstone_error(error):
...
+
+    if (
+        record
+        and record.get("access", {}).get("record", None) == "restricted"
+        and "doi" in record.get("pids", {})
+    ):
+        return (
+            render_template(
+                "invenio_app_rdm/records/restricted_tombstone.html",
```
Comment (kpsherva) [inline invenio_app_rdm/records_ui/views/records.py:429]: very minor: should it be configurable like the 403 below?

File: invenio_app_rdm/records_ui/views/records.py [hunk 4]
```diff
@@ -412,8 +412,24 @@ def record_tombstone_error(error):
...
         # trigger the flask-login unauthorized handler
         return current_app.login_manager.unauthorized()
+
+    record = getattr(error, "record", None)
+
+    if (
+        record
+        and record.get("access", {}).get("record", None) == "restricted"
+        and "doi" in record.get("pids", {})
+    ):
```
Comment (kpsherva) [inline invenio_app_rdm/records_ui/views/records.py:426]: for readability:
```suggestion

    is_restricted = record and record.get("access", {}).get("record", None) == "restricted"
    has_doi = "doi" in record.get("pids", {})
    if is_restricted and has_doi:
```

File: invenio_app_rdm/records_ui/templates/semantic-ui/invenio_app_rdm/records/deposit.html [hunk 5]
```diff
@@ -41,6 +41,15 @@
            value='{{ files | tojson }}'>
   {%- endif %}
 
+  {%- if config %}
+    <input type="hidden" name="deposits-groups-enabled"
+           value='{{ config.USERS_RESOURCES_GROUPS_ENABLED | tojson }}'>
```
[the remark this answers is not in the document]
Comment (anikachurilova) [reply invenio_app_rdm/records_ui/templates/semantic-ui/invenio_app_rdm/records/deposit.html:46]: victim of a rebase, i will remove it

**Pull request merged**